dist/js/multiselect.js

```diff
@@ -1,755 +1,1812 @@
+             */
+            function isSafariBrowser() {
+                /** The user agent string of the browser.
+                 * @type {string} */
+                var ua = window.navigator.userAgent;
+                return (ua.toLowerCase().indexOf(USER_AGENT_SAFARI) > -1);
+            }
+
+            /**
+             * This provides lexicographic comparison of two options or optgroups,
+             * i.e. the only relevant elements in the multiselect.
+             * Examples:
+             * 100 &lt; 99
+             * abc100 &lt; abc99
+             * aaa &lt; bbb
+             * @param {HTMLOptionElement|HTMLOptGroupElement} a - the first element
+             * @param {HTMLOptionElement|HTMLOptGroupElement} b - the second element
+             * @returns {COMPARISON_RESULTS} - the result of the comparison
+             */
+            function lexicographicComparison(a, b) {
+                /** The text used to compare item a
+                 * @type {string} */
+                var aText = getOptionOrOptgroupText(a);
+                /** The text used to compare item b
+                 * @type {string} */
+                var bText = getOptionOrOptgroupText(b);
+                if (aText === "" || bText === "") {
+                    return COMPARISON_RESULTS.DO_NOT_CHANGE;
+                }
+                // lexicographic comparison between strings (compare chars at same index)
+                return (aText < bText) ? COMPARISON_RESULTS.A_COMES_FIRST : COMPARISON_RESULTS.B_COMES_FIRST;
+            }
+
+            /**
+             * Takes an option or optgroup and returns the appropriate text to use for comparison.
+             * @param {HTMLOptionElement|HTMLOptGroupElement} optionOrOptgroup - the HTML element to check
+             * @returns {string} - the text for the element (if it is an option or optgroup)
+             */
+            function getOptionOrOptgroupText(optionOrOptgroup) {
+                /** The element tag name
+                 * @type {string} */
+                var tagName = optionOrOptgroup.tagName;
+                if (tagName == "OPTION") {
+                    return optionOrOptgroup.innerHTML;
+                }
+                if (tagName == "OPTGROUP") {
+                    return optionOrOptgroup.label;
+                }
+                return "";
+            }
+
+            /**
+             * Comparison using the initial position of elements
+             * stored during the initialization process.
+             * @param {HTMLOptionElement|HTMLOptGroupElement} a - the first element
+             * @param {HTMLOptionElement|HTMLOptGroupElement} b - the second element
+             * @returns {COMPARISON_RESULTS} - the result of the comparison
+             */
+            function initialOrderComparison(a, b) {
+                /** The initial index for a
+                 * @type {number} */
+                var aPositionIndex = getInitialPosition($(a));
```

Comment:
I wrapped the position getter/setter in a function because then it would be easier to change.